community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/legacy/LuceneDataSource.java

```diff
@@ -281,6 +325,15 @@ synchronized IndexReference syncGetIndexSearcher( IndexIdentifier identifier )
             {
                 indexReference = indexReferenceFactory.createIndexReference( identifier );
                 indexSearchers.put( identifier, indexReference );
+                HashMap<String,DocValuesType> fieldTypes = new HashMap<>();
+                for ( LeafReaderContext leafReaderContext : indexReference.getSearcher().getTopReaderContext().leaves() )
+                {
+                    for ( FieldInfo fieldInfo : leafReaderContext.reader().getFieldInfos() )
+                    {
+                        fieldTypes.put( fieldInfo.name, fieldInfo.getDocValuesType() );
```

Comment:
also not quite sure when we should remove field type info from map. Would expect that in normal circumstances people do not have lots dynamic fields in index, but maybe some point for future

Reply:
Since lucene only cleans the internal map when removing the index entirely, we should to the same.